pkg/elle/core/graph.go

```diff
@@ -19,75 +19,263 @@ type DirectedGraph struct {
 
 // Vertices returns the set of all vertices in graph
 func (g *DirectedGraph) Vertices() []Vertex {
-	panic("impl me")
+	vertices := []Vertex{}
+
+	for key := range g.Outs {
+		vertices = append(vertices, key)
+	}
+	return vertices
 }
 
-// In returns inbound edges to v in graph g
+// In returns inbound vertices to v in graph g
 func (g *DirectedGraph) In(v Vertex) []Vertex {
-	panic("impl me")
+	_, ok := g.Ins[v]
+	if !ok {
+		return []Vertex{}
```

Comment:
So, will we meet the situation that v isn't in `Ins`?

Reply:
`return nil` is posible.

Reply:
If the vertices are normal, this situation will not be encountered, otherwise it will return nil.